setup.py

```diff
@@ -4,6 +4,37 @@
 
 from setuptools import find_packages, setup
 
+# Fail with explicit error if trying to install Django 2 on Python 2, and
+# advise on how to avoid this.
+if sys.version_info < (3,):
+    sys.stderr.write(
+        """
+
+========
+WARNING!
+========
+
+You are running Django >= 2 on Python 2, but it requires Python 3.
+
+Unfortunately Django 2.0 and above are not compatible with Python 2,
+but you still ended up trying to install this version.  This should
```

Comment:
I'm not sure "This should not have happened."  adds much.
Also, use 1 space between sentences.